### PR TITLE
feat(geometry): a manifold accept gate for boolean results, feature-gated, with the census that says why it stays off

### LINKED ISSUE #3440: Every boolean op's accept path validates finiteness and index bounds only, so a torn kernel result is accepted silently
Every boolean operation validates its kernel output through `ClippingProcessor::validate_mesh` before accepting it. That function checks finiteness and index bounds and nothing else, so a torn result is accepted silently — including on paths whose own comments claim watertightness as part of the contract.

This is separate from #3435 (which is the tear itself). This is about why a tear can be accepted without anything noticing, and it should be fixed on its own terms whatever the root cause of #3435 turns out to be.

## The gap

```rust
/// Validate mesh for common issues
fn validate_mesh(&self, mesh: &Mesh) -> bool {
    if mesh.positions.iter().any(|v| !v.is_finite()) { return false; }
    if mesh.normals.iter().any(|v| !v.is_finite()) { return false; }
    let vertex_count = mesh.vertex_count();
    for idx in &mesh.indices {
        if *idx as usize >= vertex_count { return false; }
    }
    true
}
```

No topology check of any kind. It cannot observe an open edge.

**Four call sites, all in `csg/mod.rs`, none elsewhere in the repo** — so this covers every boolean op, not one path:

| line | operation |
|---|---|
| 328 | `subtract_mesh` |
| 401 | `subtract_mesh_many` (per chunk) |
| 428 | `union_mesh` |
| 452 | `intersection_mesh` |

All four share the same contract: `false` → discard the kernel result, fall back, and `record_failure(..., KernelOutputInvalid)`.

**Measured consequence:** host `#43810` in `ISSUE_068_ARK_NUS_skolebygg.ifc` passes `validate_mesh` and comes out with `open=42`. `subtract_mesh_many`'s own comment states the batch path's contract as "volume parity + **watertightness**" — that half has nothing enforcing it.

## Why the obvious fix is not the first step

Adding a topology check and returning `false` on failure is a **behaviour** change, not a warning. On failure the caller discards the kernel result and falls back, so a hard gate would silently reroute every currently-torn host onto a different code path — across difference, union and intersection at once. Whether those hosts land better, worse, or merely differently is unknown, and **it cannot be determined by reading**: the census records per-host open-edge counts at the final output but no path attribution, so nothing today can answer "how many hosts would flip".

## Suggested order

**1. Diagnostic first, zero behaviour change.** Call a topology check at the four sites *without* changing `validate_mesh`'s return, recording a new `BoolFailureReason` variant through the `record_failure` channel that is already wired at all four. That channel already reaches per-host `csg_failure_count` / `first_failure_label` in `router/diagnostics.rs`, and nothing branches production geometry on it — there is precedent for informational variants (`NoBoundsOverlap` is documented as "Informational — host returned unchanged"). That yields the corpus-wide flip count using production infrastructure rather than a bespoke harness.

**2. Then gate behind a feature.** `debug_geometry` / `csg_capture` / `observability` already gate exactly this class of check in `prism_cut.rs`. Enforcing strictly in CI while production tolerates lets the census run with the gate on and produce a zero-tolerance signal before anything flips for real.

**3. Hard gate last**, once the flip set's outcomes are known.

## Do not write a fourth copy of the rule

`directed_closed` in `router/voids/prism_cut/closure_checks.rs` is exactly the check needed, and there is live precedent for using it as an accept/reject signal — `prism_cut.rs` already gates on it at five sites with its own `out_not_closed` counter. It is `pub(super)` inside a private module, so it is not visible from `csg/mod.rs` today; making it reachable is a visibility change, not new logic.

Writing a fresh "is this closed" implementation in `csg/mod.rs` would be a fourth definition of watertightness in this crate, which is precisely the divergence #3435 documents between `closed_or_hairline` and the census's `edge_stats`. A shared home should be a hard requirement of any option that adds the check.

Cost is not the obstacle: `directed_closed` is an O(triangles) hash sweep already run several times per candidate mesh in the hot prism path.

Refs #3435

### PR DESCRIPTION
**Pull request by louistrue**
## Summary

Every boolean op's accept path validated only finiteness and index bounds, so a torn result (non-manifold edges, flipped winding) was accepted as success (#3440; steps 1 and 2 landed in #3717).

- New predicate `edge_multiplicity_defects` in the shared closure-checks module counts unsigned uses per undirected edge on exact f32 coordinate keys (`-0.0` folded): `over_used` (more than two uses) and `same_direction` (an edge used twice the same way). The signed tallies the closure predicates use cancel exactly these cases. The key is exact rather than snapped on purpose: snapping is right for a closure check (wrongly closing a hairline errs toward accepting) and wrong for a rejection check (fusing two surfaces 0.03 mm apart reported `over_used: 12` on two clean bipyramids); a sub-ulp non-manifold edge is therefore a miss, not a false rejection.
- Wired at the four accept sites (`subtract_mesh`, `subtract_mesh_many`, `intersection_mesh`, `audit_and_gate_union`); a rejection falls back exactly like `KernelOutputInvalid` (uncut host, plain merge, never `Err`) and records `BoolFailureReason::NonManifoldRejected { over_used, same_direction }`. Both gates are evaluated and combined with `|`, so a combined-feature census attributes each.
- **Off by default**, behind `csg_manifold_gate`, because the 115-model census (2071 void hosts) shows 110 hosts rejected, all re-cut by the #635 AABB fallback, and that fallback is worse than the tear: `issue_098_reveal_wall` 42 to 380 unpaired edges, `issue_098_v5c` 108 to 416 and 25% more wall removed, one wall of `issue_960_segmented_roof_clip` regressing. Those numbers are pinned under the feature, the default build's assertions are untouched, and a default-build test pins the gate as a true no-op (cfg'd on neither feature, so a `csg_topology_gate`-only build is unaffected). The census asserts its counts exactly against the manifest blob it measured.
- `cargo semver-checks` against 7.1.1 fails three major checks in this unpublished window: `NonManifoldRejected` (this PR), `OpenTopologyRejected` (#3717) and `aggregate_diagnostics`'s parameter count (#3766); the release lane was already blocked by the latter two. `majorOffset` moves 2 to 3 with the measured reason and manifests re-synced, taking published crates to 9.x. That commit (bd03fae8a) is the widest change here and reverts cleanly on its own if it should be a separate PR.
- `router/diagnostics.rs` lost a verbatim second copy of `BoolFailureReason::label`'s match; its row is lowered, no row raised.

Closes #3440

## Verification

- Default build: `cargo test -p ifc-lite-geometry -p ifc-lite-processing --no-fail-fast` 1441 pass; feature build: 1130 pass; clippy `-D warnings` clean on all four feature combinations; ratchet 6/6; `check-rust-major-offset` exit 0.
- RED/GREEN per check with deliberately torn fixtures; mutation: deleting the `subtract_mesh` gate call fails `manifold_gate_rejects_a_non_manifold_result_at_the_accept_seam`; restoring the snapped key fails the bipyramid and 0.05 mm slab tests.
- Not done: a feature-on CI lane (`csg_topology_gate` is itself red on main on three targets, inherited from #3717, and needs its own pins first); filed as a follow-up.
- Changeset: `@ifc-lite/wasm` patch (nothing enables the feature).

## Summary by CodeRabbit

* **New Features**
  * Added an opt-in CSG manifoldness gate that rejects non-manifold boolean results and uses existing fallback behavior.
  * Added diagnostic reporting for rejected non-manifold results, including edge-defect counts.
  * Preserved existing behavior when the feature is disabled.

* **Bug Fixes**
  * Reduced spurious diagnostics by applying validation only to generated boolean results.

* **Tests**
  * Added coverage for manifold detection, rejection, fallback behavior, clean results, and topology edge cases.

* **Chores**
  * Updated the release version and internal package references to 9.1.1.

### CODE REVIEW

File: rust/geometry/src/csg/union.rs [hunk 1]
```diff
@@ -126,7 +126,10 @@ impl ClippingProcessor {
         if !self.validate_mesh(&mesh) {
             return mesh;
         }
-        if self.topology_gate_reject(BoolOp::Union, &mesh) {
+        // Both gates run; see `subtract_mesh` for why this is not `||`.
+        let manifold_rejected = self.manifold_gate_reject(BoolOp::Union, &mesh);
```
[the remark this answers is not in the document]
Comment (louistrue) [reply rust/geometry/src/csg/union.rs]: Confirmed and fixed in 54dfec09c. Wrote the test first: `union_mesh(finned_box, empty)` and the mirrored argument order recorded two `NonManifoldRejected { over_used: 1, same_direction: 0 }` under `csg_manifold_gate`, and two `KernelError(open topology ...)` in the default build, for a union that never ran. `union_mesh` now short-circuits to `union_pair` when either operand is empty, which is the same call the `unioned` guard in `union_meshes` already makes, and it makes the function's own "empty operands are handled silently" doc true. The returned geometry was never wrong (merging a mesh with an empty one is that mesh), so this is purely the diagnostics and census pollution you named.

I re-ran the pinned corpus census with the fix in to check the headline numbers did not move: 115 fixtures, 2071 void hosts, 110 rejected, 288/3436 over-used and 120/917 same-direction, all unchanged. Every rejection in the corpus is a `Difference`, so the union path contributes nothing to the census either way. Test is `union_with_an_empty_operand_records_no_failure`, feature-independent so it pins both channels.
